### PR TITLE
#84 - Implement deleting by name and version in IndexYaml

### LINKED ISSUE #84: Delete specified chart from index.yaml file
It is necessary to support deleting of specified chart from `index.yaml` file within `IndexYaml` class. The methods `deleteByName` and `deleteByNameAndVersion` should be implemented and tested.

### PR DESCRIPTION
**Pull request by genryxy**
Closes #84 
Implemented deleting by name and version in `IndexYaml`

### CODE REVIEW

File: src/test/java/com/artipie/helm/IndexYamlTest.java [hunk 1]
```diff
@@ -207,6 +207,42 @@ void failsToDeleteChartByNameWhenIndexYamlAbsent() {
...
+        new TestResource("index.yaml").saveTo(this.storage);
+        new IndexYaml(this.storage, IndexYamlTest.BASE)
+            .deleteByNameAndVersion(chart, "0.4.1")
+            .blockingGet();
+        final IndexYamlMapping mapping = this.mapping();
+        MatcherAssert.assertThat(
+            mapping.entries().containsKey(chart),
+            new IsEqual<>(false)
+        );
+    }
```
Comment (olegmoz) [inline src/test/java/com/artipie/helm/IndexYamlTest.java:244]: @genryxy please also add a test for case when there are no versions for specified `name`

File: src/main/java/com/artipie/helm/metadata/IndexYaml.java [hunk 2]
```diff
@@ -135,6 +157,18 @@ public Completable deleteByName(final String name) {
         return res;
     }
 
+    /**
+     * Generate exception.
+     * @return Not found exception.
+     */
+    private static Single<Map<String, Object>> notFoundException() {
```
Comment (olegmoz) [inline src/main/java/com/artipie/helm/metadata/IndexYaml.java:164]: @genryxy I'd suggest to change function's signature to `private static <T> Single<T> notFoundException()` as it does not really care about returned type

**Review by olegmoz — CHANGES_REQUESTED** [review]
@genryxy thanks, please see the comments

**Comment by genryxy** [comment]
@olegmoz thanks, added tests for trying to delete absent chart and for trying to delete absent version of chart

**Review by olegmoz — APPROVED** [review]
@genryxy thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/helm-adapter#88` is not assigned, can't get performer

**Pull request merged**